packages/components/src/AnimatedPresence/AnimatedPresence.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { Children, PropsWithChildren, useEffect } from "react";
+import { AnimatePresence, motion, useReducedMotion } from "framer-motion";
+import {
+  TIMING_BASE,
+  TIMING_QUICK,
+  fade,
+  fromBottom,
+  fromLeft,
+  fromLeftToRight,
+  fromRight,
+  fromRightToLeft,
+  fromTop,
+  popIn,
+} from "./AnimatedPresence.transitions";
+import { usePreviousValue } from "./hooks/usePreviousValue";
+
+const transitions = {
+  fromBottom,
+  fromTop,
+  fromLeft,
+  fromRight,
+  popIn,
+  fromLeftToRight,
+  fromRightToLeft,
+  fade,
+};
+
+export type AnimatedPresenceTransitions = keyof typeof transitions;
+
+interface AnimatedPresenceProps extends Required<PropsWithChildren> {
+  /**
+   * The type of transition you can use.
+   */
+  readonly transition?: AnimatedPresenceTransitions;
+
+  /**
+   * Whether or not to animate the children on mount. By default it's set to false.
+   */
+  readonly initial?: boolean;
+
+  /**
+   * If you only have 1 element visible at all times, like a setup wizard or a
+   * page, This ensures the transition between the previous and next elements
+   * doesn't overlap.
+   *
+   * Using this with multiple elements visible at the same time will cause an
+   * unexpected behavior.
+   */
+  readonly exitBeforeEnter?: boolean;
+}
+
+export function AnimatedPresence(props: AnimatedPresenceProps) {
+  const shouldReduceMotion = useReducedMotion();
+
+  if (shouldReduceMotion) {
+    return <>{props.children}</>;
```

Comment:
I like this because you know it's going to be the first question asked about this feature "How do I turn it off?" And the response is "we respect browser settings" which is perfect. Great stuff.

Reply:
yeah, also a good example of componentizing too because i could've just gone and do it in 1 component but doing it this way saves the client side a bit more logic and useEffect since those would be rendered too even if you don't want them. This way it's not gonna get to that code at all.

Reply:
Should add an accessibility section to the docs (under the template it would be an h2 near the bottom) to note that it respects `prefers-reduced-motion`, as I think designers will probably ask about this more than anyone

Reply:
oh shit you right!